src/crl.c

```diff
@@ -599,10 +603,58 @@ static void SetCrlInfoFromDecoded(DecodedCRL* entry, CrlInfo *info)
     info->nextDate = (byte *)entry->nextDate;
     info->nextDateMaxLen = MAX_DATE_SIZE;
     info->nextDateFormat = entry->nextDateFormat;
-    info->crlNumber = (sword32)entry->crlNumber;
+    XMEMCPY(info->crlNumber, entry->crlNumber, CRL_MAX_NUM_SZ);
+    info->crlNumberSet = entry->crlNumberSet;
 }
 #endif
 
+/* Returns 1 if prev crlNumber is smaller
+ *         0 if equal
+ *        -1 if prev crlNumber is larger */
+static int CompareCRLnumber(CRL_Entry* prev, CRL_Entry* curr)
+{
+    int result = 0;
+#ifdef WOLFSSL_SMALL_STACK
+    mp_int* prev_num = (mp_int*)XMALLOC(sizeof(*prev_num), NULL,
+                        DYNAMIC_TYPE_BIGINT);
+    mp_int* curr_num = (mp_int*)XMALLOC(sizeof(*curr_num), NULL,
+                        DYNAMIC_TYPE_BIGINT);
+    if (prev_num == NULL || curr_num == NULL) {
+        return MEMORY_E;
+    }
+#else
+    mp_int prev_num[1];
+    mp_int curr_num[1];
+#endif
+
+    mp_init(prev_num);
```

Comment:
Technically these have return codes... you might consider using mp_init_multi to simplify cleanup.